File: Cpp/0453_minimum_moves_to_equal_array_elements.cpp
// https://leetcode.cn/problems/minimum-moves-to-equal-array-elements/

class Solution {
public:
    int minMoves(vector<int>& nums) {
        // Increasing n-1 elements by 1 works similarly as decreasing 1 element by 1.
        int min_ = nums[0], sum_ = 0;
        for (int num: nums) {
            if (num < min_) min_ = num;
            sum_ += num;
        }
        int res = sum_ - min_ * nums.size();
        return res;
    }
};

File: Python3/0453_minimum_moves_to_equal_array_elements.py
# https://leetcode.cn/problems/minimum-moves-to-equal-array-elements/

class Solution:
    def minMoves(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: int
        """

        # Increasing n-1 elements by 1 works similarly as decreasing 1 element by 1.
        min_ = min(nums)
        res = 0
        for num in nums:
            res += (num - min_)
        return res